v3/testvectors/compatibility_test.go

```diff
@@ -597,3 +598,78 @@ func TestEnableLegacyDecryptBothFormats(t *testing.T) {
 		t.Errorf("expect %v text, got %v", e, a)
 	}
 }
+
+func TestUnicodeEncryptionContextV3(t *testing.T) {
+	rune128 := string(rune(128))
+	rune200 := string(rune(200))
+	rune256 := string(rune(256))
+	runeMaxInt := string(rune(math.MaxInt32))
+	shorter := "我"
+	medium := "Brøther, may I have the lööps"
+	longer := "我的资我的资源我的资源我的资源的资源源"
+	mix := "hello 我的资我的资源我的资源我的资源的资源源 goodbye"
+	mixTwo := "hello 我的资我的资源我的资源我的资源的资源源 goodbye我的资"
+
+	unicodeStrings := []string{rune128, rune200, rune256, runeMaxInt, shorter, medium, longer, mix, mixTwo}
+	for _, s := range unicodeStrings {
+		UnicodeEncryptionContextV3(t, s)
+	}
+}
+
+func UnicodeEncryptionContextV3(t *testing.T, metadataString string) {
```

Comment:
Is it possible to have a negative test where this fails?

Reply:
Hm, good point, but I can't think of what that would entail here. As far as we know, the double-encoding is a bijection to the original string, so there aren't any known cases where we fail to decode. The only way to induce failure I can think of would be something convoluted like overwriting the object metadata out-of-band.